MapboxDirections/MBDirections.swift

```diff
@@ -128,14 +131,26 @@ open class Directions: NSObject {
      - parameter host: An optional hostname to the server API. The [Mapbox Directions API](https://www.mapbox.com/api-documentation/?language=Swift#directions) endpoint is used by default.
      */
     public init(accessToken: String?, host: String?) {
-        let accessToken = accessToken ?? defaultAccessToken
-        assert(accessToken != nil && !accessToken!.isEmpty, "A Mapbox access token is required. Go to <https://www.mapbox.com/studio/account/tokens/>. In Info.plist, set the MGLMapboxAccessToken key to your access token, or use the Directions(accessToken:host:) initializer.")
-        
-        self.accessToken = accessToken!
-        
-        var baseURLComponents = URLComponents()
-        baseURLComponents.scheme = "https"
-        baseURLComponents.host = host ?? "api.mapbox.com"
+		var baseURLComponents: URLComponents
+        if host == nil {
```

Comment:
It'd be cleaner to do a guard let here.

Reply:
I quite disagree. It's a pure if else. There's nothing to guard against.

Reply:
You're guarding against assuming [`host`](https://github.com/mapbox/MapboxDirections.swift/pull/198/files/d2cce68548e431d48838dedc3e326fda5feec0a8#diff-e76be193f7c380a503afe0e47c1cd829R145) is not nil and [also this assert](https://github.com/mapbox/MapboxDirections.swift/pull/198/files/d2cce68548e431d48838dedc3e326fda5feec0a8#diff-e76be193f7c380a503afe0e47c1cd829R150). IMO, asserting is great time to throw in a guard.

Reply:
@RomainQuidet upon reviewing the code, I understand your intent. Probably it would be more clear if you safely unwrapped the `host` variable like so: 
```
if let host = host {
   var urlString = host
   if host.hasPrefix("http") == false {
      urlString = "http://\(urlString)"
   }
   let url = URL(string: urlString)
   assert(url != nil, "A valid custom host must be provided (ex: my_host.com or full http://my_host.com:8080")
   baseURLComponents = URLComponents(url: url!, resolvingAgainstBaseURL: false)!
} else {
   let accessToken = accessToken ?? defaultAccessToken
   assert(accessToken != nil && !accessToken!.isEmpty, "A Mapbox access token is required. Go to <https://www.mapbox.com/studio/account/tokens/>. In Info.plist, set the MGLMapboxAccessToken key to your access token, or use the Directions(accessToken:host:) initializer.") 
   self.accessToken = accessToken
   baseURLComponents = URLComponents()
   baseURLComponents.scheme = "https"
   baseURLComponents.host = mapboxHost
}
```